assertion-analyser.js

```diff
@@ -1,103 +1,119 @@
 function objParser(str, init) {
   // finds objects, arrays, strings, and function arguments
   // between parens, because they may contain ','
-  var openSym = ['[', '{', '"', "'", '('];
-  var closeSym = [']', '}', '"', "'", ')'];
-  var type;
-  for(var i = (init || 0); i < str.length; i++ ) {
+  const openSym = ["[", "{", '"', "'", "("];
+  const closeSym = ["]", "}", '"', "'", ")"];
+  let type;
+  for (var i = init || 0; i < str.length; i++) {
     type = openSym.indexOf(str[i]);
-    if( type !== -1)  break;
+    if (type !== -1) break;
   }
   if (type === -1) return null;
-  var open = openSym[type];
-  var close = closeSym[type];
-  var count = 1;
-  for(var k = i+1; k < str.length; k++) {
-    if(open === '"' || open === "'") {
-      if(str[k] === close) count--;
-      if(str[k] === '\\') k++;
+  const open = openSym[type];
+  const close = closeSym[type];
+  let count = 1;
+  for (var k = i + 1; k < str.length; k++) {
+    if (open === '"' || open === "'") {
+      if (str[k] === close) count--;
+      if (str[k] === "\\") k++;
     } else {
-      if(str[k] === open) count++;
-      if(str[k] === close) count--;
+      if (str[k] === open) count++;
+      if (str[k] === close) count--;
     }
-    if(count === 0) break;
+    if (count === 0) break;
   }
-  if(count !== 0) return null;
-  var obj = str.slice(i, k+1);
+  if (count !== 0) return null;
+  const obj = str.slice(i, k + 1);
   return {
-    start : i,
+    start: i,
     end: k,
-    obj: obj
+    obj: obj,
   };
 }
 
 function replacer(str) {
   // replace objects with a symbol ( __#n)
-  var obj;
-  var cnt = 0;
-  var data = [];
-  while(obj = objParser(str)) {
+  let obj;
+  let cnt = 0;
+  let data = [];
+  // obj = objParser(str) is not a syntax error
+  while ((obj = objParser(str))) {
```

Comment:
One too many sets of parentheses?

Reply:
The prettier/linter did that... I am not sure why, but I tend not to question such.